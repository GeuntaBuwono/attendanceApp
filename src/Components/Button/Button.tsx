/* eslint-disable react-native/no-inline-styles */
import Label from 'Components/Text/Text'
import { TouchableOpacity } from 'react-native'
import styled from 'styled-components'
import Colors from 'Styles/colors'

import { ButtonInterface, ButtonStyleInterface } from './Button.type'

const ButtonStyled = styled(TouchableOpacity)<ButtonStyleInterface>`
  background-color: ${(props) => (props.color === 'topaz' ? Colors.topaz : Colors.silver)};
  border-radius: 6px;
  padding: ${(props) => props.padding};
`

const Button = ({ label, onPress, color, padding }: ButtonInterface) => (
  <ButtonStyled onPress={onPress} color={color} padding={padding ?? '12px'}>
    <Label restStyle={{ color: Colors.white, textAlign: 'center' }}>{label}</Label>
  </ButtonStyled>
)

export default Button
