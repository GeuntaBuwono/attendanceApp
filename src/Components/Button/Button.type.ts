export interface ButtonStyleInterface {
  color: 'topaz' | 'grey'
  padding?: string
}

export interface ButtonInterface extends ButtonStyleInterface {
  label: string
  onPress: () => void
}
