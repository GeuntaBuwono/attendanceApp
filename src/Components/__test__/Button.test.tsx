/**
 * @format
 */

import 'react-native'

import { render } from '@testing-library/react-native'
import Button from 'Components/Button/Button'
import React from 'react'
// Note: test renderer must be required after react-native.

describe('Button', () => {
  it('Render Correctly', async () => {
    const press = jest.fn()
    const tree = render(<Button label="Button Test" onPress={() => press()} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
