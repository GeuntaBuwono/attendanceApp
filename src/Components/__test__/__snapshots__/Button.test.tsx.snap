// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render Correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    className="sc-bczRLJ bLKfFM"
    color="#2C2C2B"
    style={
      Object {
        "color": "#FFF",
        "textAlign": "center",
      }
    }
  >
    Button Test
  </Text>
</View>
`;
